rbx_types/src/basic_types.rs

```diff
@@ -74,10 +76,45 @@ pub struct Vector3 {
     pub z: f32,
 }
 
+fn approx_unit_or_zero(value: f32) -> Option<i32> {
+    if abs_diff_eq!(value.abs(), 1.0) {
+        if value.is_sign_positive() {
+            Some(1)
+        } else {
+            Some(-1)
+        }
+    } else if abs_diff_eq!(value.abs(), 0.0) {
+        Some(0)
+    } else {
+        None
+    }
+}
+
+fn get_normal_id(position: u8, value: i32) -> Option<u8> {
```

Comment:
This feels rather cryptic - is there a chance we could get some documentation for it in-file?